File: OOP(Object-Oriented Programing/CraneSystem.java
import java.util.*;

abstract class Position {
    int x;
    int y;

    Position() {
        this.x = 0;
        this.y = 0;
    }

    void moveX(int dx) { this.x += dx; }
    void moveY(int dy) { this.y += dy; } 
}

class Motor extends Position {
    private boolean isActive;

    Motor() {
        isActive = false;
    }

    public void activate() {
        isActive = true;
    }

    public void deactivate() {
        isActive = false;
    }

    public boolean isActive() {
        return isActive;
    }
}

class Box extends Position {
    int z;

    Box() {
        z = 0;
    }

    void move(int dx, int dy) {
        this.x += dx;
        this.y += dy;
    }
}

class CraneSystem {
    ArrayList<Motor> motors = new ArrayList<>();
    Box object = new Box();

    ArrayList<String> commands = new ArrayList<>();

    CraneSystem() {
        for(int i = 0; i < 8; i++) {
            motors.add(new Motor());
        }
    }

    void moveBox(int dx, int dy) {
        object.move(dx, dy);
    }

    boolean isBoxAt(int x, int y) {
        return object.x == x && object.y == y;
    }


public static void main(String[] args) throws Exception {
    CraneSystem cs = new CraneSystem();
    Scanner scanner = new Scanner(System.in);

    while(true) {
        System.out.print("Enter a command: ");
        String command = scanner.nextLine();

        boolean flag = false;

        System.out.print("Enter a command: ");
        String command = scanner.nextLine();

        boolean flag = false;

        if(command.startsWith("move")) {
            String[] parts = command.split(" ");
            int dx = Integer.parseInt(parts[1]);
            int dy = Integer.parseInt(parts[2]);
            cs.moveBox(dx, dy);
            cs.commands.add(command);
        }
        else if(command.startsWith("check")) {
            String[] parts = command.split(" ");
            int x = Integer.parseInt(parts[1]);
            int y = Integer.parseInt(parts[2]);
            if(cs.isBoxAt(x, y)) {
                System.out.println("The box is at (" + x + ", " + y + ").");
            } else {
                System.out.println("The box is not at (" + x + ", " + y + ").");
            }
        }

        if(command.equals("right")) {
            cs.motors.get(1-1).activate();
            cs.motors.get(1-1).moveX(1);
            cs.commands.add(command);  
        }
        else if(command.equals("left")) {
                cs.motors.get(1-1).activate();
                cs.motors.get(1-1).moveX(-1);
                cs.commands.add(command);
        }
        else if(command.equals("down")) {
            cs.motors.get(2-1).activate();
            cs.motors.get(2-1).moveY(-1);
            cs.commands.add(command); 
        }
        else if(command.equals("up")) {
            cs.motors.get(2-1).activate();
            cs.motors.get(2-1).moveY(-1);
            cs.commands.add(command);
        }
        else if(command.equals("grab")) {
            boolean allStragiht = true;
            boolean allInactive = true;

            for (int i = 4; i <= 8; i ++) {
                if (cs.motors.get(i-1).isActive) {
                    allInactive = false;
                    break;
                }
            }

            if (allStragiht && allInactive) {
                for (int i = 4; i <= 8; i ++) {
                    cs.motors.get(i-1).activate();
                }

                cs.object.z += 10;
                cs.commands.add(command);
            } else {
                System.out.println("Cannot grab. All or some of the fingers are already bent or some motors are active.");
            }
        }
        else if(command.equals("letgo")) {
            boolean allActivate = true;

            for(int i = 4; i <= 8; i++) {
                if (!cs.motors.get(i-1).isActive) {
                    allActivate = false;
                    break;
                }
            }

            if(allActivate) {
                for(int i = 4; i <= 8; i++) {
                    cs.motors.get(i-1).deactivate();
                }

                cs.object.z -= 10;
                cs.commands.add(command);
            }
        
         else{
              System.out.println("Cannot let go. Some of the motors are not active.");
           }
        }

        else if (command.equals("exit")) {
            System.out.println("End the crane operation");
            break;
        } else {
            System.out.println("Invalid command! Please enter again");
            flag = true;
        }
    
        if (!flag) {
            System.out.println("command List: ");
            for (int i = 0; i < cs.commands.size(); i++) {
                System.out.println((i)+"."+cs.commands.get(i));
            }
            System.out.println("---------------");
        }
    }
}
}

File: OOP(Object-Oriented Programing/CraneSystem.cpp
#include <iostream>
#include <vector>
#include <string>

class Position {
private:
    int x, y;

public:
    Position() : x(0), y(0) {}

    void moveX(int dx) { x += dx; }
    void moveY(int dy) { y += dy; }
    int getX() const { return x; }
    int getY() const { return y; }
};

class Motor : public Position {
private:
    bool isActive;

public:
    Motor() : isActive(false) {}

    void activate() { isActive = true; }
    void deactivate() { isActive = false; }
    bool getIsActive() const { return isActive; }
};

class Box : public Position {
public:
    int z;

    Box() : z(0) {}
};

class CraneSystem {
private:
    std::vector<Motor> motors;
    Box object;
    bool isGrabbed;
    static std::vector<std::string> commands;

public:
    CraneSystem() {
        for (int i = 0; i < 8; i++) {
            motors.push_back(Motor());
        }
    }

    static void addCommand(const std::string& command) {
        commands.push_back(command);
    }

    static const std::vector<std::string>& getCommands() {
        return commands;
    }

    static void printCommands() {
        std::cout << "Command List: \n";
        for (size_t i = 0; i < commands.size(); i++) {
            std::cout << i + 1 << ". " << commands[i] << "\n";
        }
        std::cout << "———————\n";
    }

    void printPositions() const {
        std::cout << "Crane Motor1 Position: (X: " << motors[0].getX() << ", Y: " << motors[0].getY() << ")\n";
        std::cout << "Crane Motor2 Position: (X: " << motors[1].getX() << ", Y: " << motors[1].getY() << ")\n";
        std::cout << "Box Position: (X: " << object.getX() << ", Y: " << object.getY() << ", Z: " << object.z << ")\n";
        std::cout << "———————\n";
    }

    void craneCommand(const std::string& command) {
        if (command == "right") {
            motors[0].activate();
            motors[0].moveX(1);
            if (isGrabbed) object.moveX(1);
        } else if (command == "left") {
            motors[0].activate();
            motors[0].moveX(-1);
            if (isGrabbed) object.moveX(-1);
        } else if (command == "down") {
            motors[1].activate();
            motors[1].moveY(-1);
            if (isGrabbed) object.moveY(-1);
        } else if (command == "up") {
            motors[1].activate();
            motors[1].moveY(1);
            if (isGrabbed) object.moveY(1);
        } else if (command == "grab") {
            if (motors[0].getX() != object.getX() || motors[0].getY() != object.getY()) {
                std::cout << "Cannot grab.\n";
                return;
            }

            bool allInactive = true;
            for (int i = 3; i < 8; i++) {
                if (motors[i].getIsActive()) {
                    allInactive = false;
                    break;
                }
            }
            if (allInactive) {
                for (int i = 3; i < 8; i++) {
                    motors[i].activate();
                }
                object.z += 10;
                isGrabbed = true;
            } else {
                std::cout << "Cannot grab. Some of the motors are active.\n";
            }
        } else if (command == "letgo") {
            bool allActive = true;
            for (int i = 3; i < 8; i++) {
                if (!motors[i].getIsActive()) {
                    allActive = false;
                    break;
                }
            }
            if (allActive) {
                for (int i = 3; i < 8; i++) {
                    motors[i].deactivate();
                }
                object.z -= 10;
                isGrabbed = false;
            } else {
                std::cout << "Cannot let go. Some of the motors are not active.\n";
            }
        } else if (command == "exit") {
            std::cout << "End the crane operation\n";
        } else {
            std::cout << "Invalid command! Please enter again\n";
        }

        addCommand(command);
        printCommands();
        printPositions();
    }
};

std::vector<std::string> CraneSystem::commands;

int main() {
    std::ios::sync_with_stdio(false);
    std::cin.tie(nullptr);

    CraneSystem cs;
    std::string command;

    while (true) {
        std::cout << "Enter a command: " << std::flush;
        std::getline(std::cin, command);
        cs.craneCommand(command);
        if (command == "exit") {
            break;
        }
    }

    return 0;
}

모터 1이 x축을 담당하고 모터 2가 y축을 담당해서 움직입니다.
인형 뽑기 기계처럼 작동하게 하였고 따라서 z축이 높이가 됩니다./
